packages/EasyCore/tests/Bridge/Symfony/Valiator/Constraints/AbnValidatorTest.php

```diff
@@ -0,0 +1,307 @@
+            'integer' => [53004085616, 'Expected argument of type "string", "integer" given'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideValidABN(): array
+    {
+        return [
+            'Valid ABN #1' => ['53004085616'],
+            'Valid ABN #2' => ['28043145470'],
+            'Valid ABN #3' => ['53004085616'],
+            'Valid ABN #4' => ['91724684688'],
+            'Valid ABN #5' => ['10000000000'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideValidEmptyValues(): array
+    {
+        return [
+            'Empty string' => [''],
+            'Null value' => [null],
+        ];
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function provideInvalidABN(): array
+    {
+        return [
+            'Invalid ABN #1' => ['10043145470'],
+            'Invalid ABN #2' => ['53004085615'],
+            'Invalid ABN #3' => ['53004085615'],
+            'Invalid ABN #4' => ['10000000001'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideValidABN
+     *
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
+     */
+    public function testValidateSucceedsWithValidAbn(string $abn): void
+    {
+        $validator = new AbnValidator();
+        $constraint = new Abn();
+        $context = $this->mockExecutionContextWithoutCalls();
+        $validator->initialize($context);
+
+        $validator->validate($abn, $constraint);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
+     */
+    public function testValidateSucceedsWithObjectToString(): void
+    {
+        $class = new class() {
+            public function __toString()
+            {
+                return '53004085616'; // Valid ABN
+            }
+        };
+        $validator = new AbnValidator();
+        $constraint = new Abn();
+        $context = $this->mockExecutionContextWithoutCalls();
+        $validator->initialize($context);
+
+        $validator->validate($class, $constraint);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * @dataProvider provideValidEmptyValues
+     *
+     * @param mixed $abn
+     *
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
+     */
+    public function testValidateSucceedsWithEmptyValue($abn): void
+    {
+        $validator = new AbnValidator();
+        $constraint = new Abn();
+        $context = $this->mockExecutionContextWithoutCalls();
+        $validator->initialize($context);
+
+        $validator->validate($abn, $constraint);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
+     */
+    public function testValidateThrowsUnexpectedTypeException(): void
+    {
+        $validator = new AbnValidator();
+        $constraint = new NotNull();
```

Comment:
Let's use `stdClass` here.

Reply:
We will get an error, `AbnValidator expected instance of Type, got stdClass`, something like this.

Reply:
Got it. Let's create anonymous class extending Constraint then. It's better not to tie with another constraint in the test.

Reply:
Done.